src/app/data-model-provider/OperationTypes.h

```diff
@@ -99,6 +100,7 @@ struct InvokeRequest : OperationRequest
 {
     ConcreteCommandPath path;
     BitFlags<InvokeFlags> invokeFlags;
+    FabricIndex accessingFabricIndex = kUndefinedFabricIndex;
```

Comment:
I do have to ask: why do we need this if we have a subject descriptor member already?  What we should have is a getter for the accessing fabric index, which gets it from our subject descriptor, instead of having duplicated members.

Reply:
CommandHandlerExchangeInterface has separate `GetSubjectDescriptor` and `GetAccessingFabricIndex` so they seem to be not coupled.

I could not follow and ensure that the subject descriptor fabricIndex is the `accessingFabricIndex`. I would very much prefer that, however making that change is probably a separate PR by itself. Unfortunately the methods were split and unsplitting them became harder.

We should also figure out what we want our code to look like. I seem to remember when I tried to remove a `GetAccessingFabricIndex` from https://github.com/project-chip/connectedhomeip/blob/master/src/app/AttributeValueEncoder.h#L155 (which is clearly just a getter from the descriptor) I was told that this is a spec term and I should explicitly keep the extra method. I think this resulted in the virtual `GetSubjectDescriptor` that is separate from `AccessingFabricIndex` and now are harder to clean out.

Is it guaranteed that GetAccessingFabricIndex() always is == to SubjectDescriptor.fabricIndex? If so maybe we can drop some of the virtual bits....

Reply:
> CommandHandlerExchangeInterface has separate GetSubjectDescriptor and GetAccessingFabricIndex 

The latter is a convenience API that lets people easily do the spec operation of "get the accessing fabric index" without messing it up by trying to understand subject descriptors in contexts where they are not relevant.

They are _absolutely_ coupled, and the subject descriptor fabric index _is_ the accessing fabric index.

>  I was told that this is a spec term and I should explicitly keep the extra method

Yes, because it makes it easier to correctly implement spec algorithms that don't mention the subject descriptor but do talk about the "accessing fabric index".

> Is it guaranteed that GetAccessingFabricIndex() always is == to SubjectDescriptor.fabricIndex?

Yes.

Reply:
Updated. I will create a separate PR enforcing that `GetAccessingFabricIndex` is not a virtual call if we have a GetSubjectDescriptor